checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1703 @@
+                        enclosingCmcfValue, enclosingMethodTree, checker);
+              } catch (JavaExpressionParseException e) {
+                // Do not issue an error here, because it would be a duplicate.
+                // The error will be issued by the Transfer class of the checker,
+                // via the CreatesMustCallForElementSupplier interface.
+                enclosingTarget = null;
+              }
+
+              if (representSame(target, enclosingTarget)) {
+                // This satisifies case 3.
+                validInvocation = true;
+              }
+            }
+          }
+        }
+      }
+      if (!validInvocation) {
+        missing.add(target);
+      }
+    }
+
+    if (missing.isEmpty()) {
+      // all targets were valid
+      return;
+    }
+
+    String missingStrs = StringsPlume.join(", ", missing);
+    checker.reportError(node.getTree(), "reset.not.owning", missingStrs);
+  }
+
+  /**
+   * Checks whether the two JavaExpressions are the same. This is identical to calling equals() on
+   * one of them, with two exceptions: the second expression can be null, and "this" references are
+   * compared using their underlying type. (ThisReference#equals always returns true, which is
+   * probably a bug and isn't accurate in the case of nested classes.)
+   *
+   * @param target a JavaExpression
+   * @param enclosingTarget another, possibly null, JavaExpression
+   * @return true iff they represent the same program element
+   */
+  private boolean representSame(JavaExpression target, @Nullable JavaExpression enclosingTarget) {
+    if (enclosingTarget == null) {
+      return false;
+    }
+    if (enclosingTarget instanceof ThisReference && target instanceof ThisReference) {
+      return enclosingTarget.getType().toString().equals(target.getType().toString());
+    } else {
+      return enclosingTarget.equals(target);
+    }
+  }
+
+  /**
+   * Given a node representing a method or constructor call, checks that if the call has a non-empty
+   * {@code @MustCall} type, then its result is pseudo-assigned to some location that can take
+   * ownership of the result. Searches for the set of same resources in {@code obligations} and adds
+   * the new LocalVarWithTree to it if one exists. Otherwise creates a new set.
+   *
+   * @param obligations the currently-tracked obligations. This is always side-effected: an
+   *     obligation is either modified to include a new resource alias (the result of the invocation
+   *     being tracked) or a new resource alias set (i.e. obligation) is created and added.
+   * @param node the node whose result is to be tracked
+   */
+  private void trackInvocationResult(Set<ImmutableSet<LocalVarWithTree>> obligations, Node node) {
+    Tree tree = node.getTree();
+    // We need to track the result of the call iff there is a temporary variable for the call node
+    // (because we only create temporaries for expressions that actually have must-call values).
```

Comment:
This is useful information.  It could be put in the overview comment, where I asked a question about when temporaries are created.

Reply:
Unfortunately it's also wrong. I corrected it (it should have said we only create temporaries for expressions that *could* have a must-call obligation, not that actually do) and added a (hopefully) more helpful note in that other location.